### ISSUE TITLE
How can i make extracted_features by myself? 

### ISSUE THREAD

**Issue by naeun0620** [RameenAbdal/CLIP2StyleGAN#2]
Hi, I am wondering how to make extracted_features by myself. 

I am guessing I can refer to code in `compute_pca.py`

```
   image_data = natsorted(glob.glob(args.path))
   with torch.no_grad():
      for i in range(len(image_data)):
         if i == 0:
            i_x = preprocess(Image.open(image_data[i])).unsqueeze(0).to(device)
            image_features_pred = model.encode_image(i_x)
         else:
            i_x = preprocess(Image.open(image_data[i])).unsqueeze(0).to(device)
            image_features_pred = torch.cat((image_features_pred, model.encode_image(i_x)),dim = 0)
```

Can i just save `image_features_pred` here as npy and use it as extracted_features, or should i post-process that?

**Comment by RameenAbdal** [comment]
Yes, you can use this to extract the image features.

**Comment by naeun0620** [comment]
Hi, I have been running through your code, and got some more questions. 

In the lines https://github.com/RameenAbdal/CLIP2StyleGAN/blob/main/optimize_dir.py#L36-L40 , 
```  
  if args.pca_axis == 0:
        predictions = ['kids', 'smiling']

    elif args.pca_axis == 19:
        predictions = ['A person with a beard', 'A person with glasses']
```
Where are the words `kids, smiling` and `A person with a beard, A person with glasses` coming from? 

Are the label outputs of `text_optimize.py`? 
If they are, when the input path is set as `PCA_FFHQ/0/*.png`, then the label outputs are children, granddaughter, child, .... I don't see the word `smiling` and `kids` precisely. 

I wonder if you set the words arbitrarily.

Thanks.
